toolchain/check/testdata/facet/facet_assoc_const.carbon

```diff
@@ -0,0 +1,268 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// INCLUDE-FILE: toolchain/testing/min_prelude/facet_types.carbon
+// EXTRA-ARGS: --dump-sem-ir-ranges=only
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/facet/facet_assoc_const.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/facet/facet_assoc_const.carbon
+
+// --- success.carbon
+library "[[@TEST_NAME]]";
+
+interface I { let T:! type; }
+
+fn F(T:! I where .T = {}) {}
+
+// --- success_associated.carbon
+library "[[@TEST_NAME]]";
+
+interface I { let T:! type; let U:! type; }
+
+fn F(T:! I where .T = .U) {}
+
+// --- fail_two_different.carbon
+library "[[@TEST_NAME]]";
+
+interface L { let W:! type; }
+
+// CHECK:STDERR: fail_two_different.carbon:[[@LINE+4]]:10: error: associated constant `.(L.W)` given two different values `{}` and `()` [AssociatedConstantWithDifferentValues]
```

Comment:
Stray thought here.

If a constant is given a different value twice, it's going to diagnose and the 2nd is marked invalid, right?

If a constant is given a different value three times, the 2nd is marked bad so it doesn't object when it finds the third. Will it re-sort and diagnose "two different values" a 2nd time for the 1st and 3rd value?

Can you test what happens if it's given 4 different values? I'm probably more confident that'll diagnose at least twice. Maybe should only diagnose once. :)